local-volume/provisioner/pkg/util/volume_util.go

```diff
@@ -131,37 +131,48 @@ func (u *volumeUtil) GetFsCapacityByte(fullPath string) (int64, error) {
 	return capacity, err
 }
 
-// GetBlockSpaceByte returns  capacity in bytes of a block device.
+// GetBlockCapacityByte returns  capacity in bytes of a block device.
 // fullPath is the pathname of block device.
-func (u *volumeUtil) GetBlockSpaceByte(fullPath string) (uint64, error) {
+func (u *volumeUtil) GetBlockCapacityByte(fullPath string) (int64, error) {
 	file, err := os.OpenFile(fullPath, os.O_RDONLY, 0)
 	if err != nil {
 		return 0, err
 	}
 	defer file.Close()
 
 	var size int64
+	// Get size of block device into 64 bit int.
+	// Ref: http://www.microhowto.info/howto/get_the_size_of_a_linux_block_special_device_in_c.html
 	if _, _, err := unix.Syscall(unix.SYS_IOCTL, file.Fd(), unix.BLKGETSIZE64, uintptr(unsafe.Pointer(&size))); err != 0 {
 		return 0, err
 	}
 
-	return uint64(size), err
+	return size, err
 }
 
 var _ VolumeUtil = &FakeVolumeUtil{}
 
 // FakeVolumeUtil is a stub interface for unit testing
 type FakeVolumeUtil struct {
 	// List of files underneath the given path
-	directoryFiles map[string][]*FakeFile
+	directoryFiles map[string][]*FakeDirEntry
 	// True if DeleteContents should fail
 	deleteShouldFail bool
 }
 
-// FakeFile contains a representation of a file under a directory
-type FakeFile struct {
-	Name     string
-	IsNotDir bool
+const (
+	// FakeEntryFile is mock dir entry of type file.
+	FakeEntryFile = "file"
+	// FakeEntryBlock is mock dir entry of type block.
+	FakeEntryBlock = "block"
+	// FakeEntryUnknown is mock dir entry of type unknown.
+	FakeEntryUnknown = "unknown"
```

Comment:
@msau42 I just realized that this is already covered in the unit test TestDiscoverVolumes_BadVolume. Let me know if any other changes are needed.